src/python/pants/backend/helm/util_rules/renderer.py

```diff
@@ -172,6 +172,9 @@ def artifacts(self) -> dict[str, FileDigest | Snapshot] | None:
     def metadata(self) -> dict[str, Any] | None:
         return {"address": self.address, "chart": self.chart, "post_processed": self.post_processed}
 
+    def cacheable(self) -> bool:
```

Comment:
Probably good to comment here why this is important.

Also, should the cacheability depend on the nature of post-processor, rather than saying that any post-processor makes things uncacheable?

Reply:
Yeah, not a big fan of this blanket approach, I sort of decided for this as a quick solution and I'm glad it's being challenged.

Because I would like to support any arbitrary runnable target, there is not much information at this point about the nature of the post-processor being used. The only way I can think of passing that information is that the targets referenced in the `post_renderers` field have a `cacheable` field. This could either be implemented generically in the `RunFieldSet` and then passed along downstream (although it most likely would have consequences to current code), or this new field could be part of yet another target, something like `helm_post_renderer`, which would be the only valid target type that can be used in the `post_renderers` field of `helm_deployment`.

Reply:
Would having a separate target open the door to pants downloading binaries like `vals`?

Reply:
IIRC there was a conversation in Slack months ago in which it was discussed to have some sort of `download_file` target type that could be used to download binaries from the Internet, sort of like `http_archive` is used in Bazel.

Having a target like I can imagine a BUILD file looking like:

```python
download_file(name="vals_bin", url="...", sha="...")

experimental_run_shell_command(name="vals_cmd", command="vals eval -f -", dependencies=[":vals_bin"])

helm_deployment(post_processors=[":vals_cmd"])
```

So the Helm backend wouldn't require any changes with current implementation to use that tool, if the `download_file` target was available.

The idea of the custom `helm_post_renderer` target would be so we can have additional info about the nature of the post-renderer tool. Like in here:

```python
pex_binary(name="custom")
experimental_run_shell_command(name="vals_cmd", command="vals eval -f -")

helm_post_renderer(name="custom_renderer", dependencies=[": custom"], cacheable=True)
helm_post_renderer(name="vals_renderer", dependencies=[":vals_cmd"], cacheable=False)

helm_deployment(post_processors=[":custom_renderer", ":vals_renderer"])
```

As you can see, this quickly creates an explosion of targets for just having an additional field, hence why I'm pondering this VS having a `cacheable` field in _runnable targets_ like `pex_binary`, `go_binary` or `experimental_run_shell_command`. This also doesn't feel right as that field would have any meaning in all the other use cases for those targets.

Reply:
Tangentially, FWIW you can today provide `http_source` as a source to a `resources` target:

```
resource(
  sources=[http_source(url="..", len="..", sha256="..")]
)
```

Reply:
So that is sort-of your `download_file` target I think?

Reply:
I think this is fine as a quick solution - but let's comment that because we have no way of knowing if the post-processor output is cacheable, we conservatively don't cache the result?

Reply:
agree, added comments to this `cacheable` method and to the process cache scope to help understand why we don't want to cache the post processors